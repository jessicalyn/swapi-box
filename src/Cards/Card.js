import React from 'react'

export default function Card({ card }) {
    return (
        <div>
            <h3>{ card.name }</h3>
            <h4>{ card.species }</h4>
            <h4>{ card.homeworld }</h4>
            <h4>{ card.homeworldPop }</h4>
        </div>
    )
}