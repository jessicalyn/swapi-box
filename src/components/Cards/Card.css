.card-body {
  margin: 2rem;
  padding: 1rem;
  border-radius: 1rem;
  font-family: 'Sawarabi Gothic', sans-serif;
  font-size: 1.5rem;
  background-color: #A5C4C9;
  color: #477480;
  border: .2rem solid #477480;
}

.fav-button {
  background-color: #BEB7A4;
}