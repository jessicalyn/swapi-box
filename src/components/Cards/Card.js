import React, { Component } from 'react'
import { connect } from 'react-redux'
import './Card.css'

export default class Card extends Component {

  clickFavorite = () => {
    const { name, favorite } = this.props.card
    console.log({favorite})
  }

  render() {
    const { card } = this.props
    return (
        <div className="card-body">
            <h3>{ card.name }</h3>
            <h4>{ card.species }</h4>
            <h4>{ card.homeworld }</h4>
            <h4>{ card.homeworldPop }</h4>
            <h4>{ card.climate }</h4>
            <h4>{ card.population }</h4>
            <h4>{ card.residents }</h4>
            <h4>{ card.terrain }</h4>
            <h4>{ card.class }</h4>
            <h4>{ card.model }</h4>
            <h4>{ card.numOfPassengers }</h4>
            <button onClick={ this.clickFavorite } className="fav-button">Favorite</button>
        </div>
    )
  }
}

// export const mapDispatchToProps = (dispatch) => ({
//   toggleFavorite: (name) => dispatch(toggleFavorite(name))
// })

// export default connect(null, mapDispatchToProps)(Card)