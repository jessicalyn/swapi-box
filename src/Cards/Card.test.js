import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme'

const mockCard = {
  name: "Naboo", 
  terrain: "grassy hills, swamps, forests, mountains", 
  population: "4500000000", 
  climate: "temperate", 
  residents: ["R2-D2", "Palpatine"]
}

describe('Card', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Card key={ mockCard.name } card={ mockCard }/>);
  });

  it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  });
})
