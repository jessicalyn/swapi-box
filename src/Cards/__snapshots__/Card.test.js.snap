// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
      Object {
        "climate": "temperate",
        "name": "Naboo",
        "population": "4500000000",
        "residents": Array [
          "R2-D2",
          "Palpatine",
        ],
        "terrain": "grassy hills, swamps, forests, mountains",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Naboo
        </h3>,
        <h4 />,
        <h4 />,
        <h4 />,
        <h4>
          temperate
        </h4>,
        <h4>
          4500000000
        </h4>,
        <h4>
          R2-D2
          Palpatine
        </h4>,
        <h4>
          grassy hills, swamps, forests, mountains
        </h4>,
        <h4 />,
        <h4 />,
        <h4 />,
        <button>
          Favorites
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Naboo",
        },
        "ref": null,
        "rendered": "Naboo",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "temperate",
        },
        "ref": null,
        "rendered": "temperate",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "4500000000",
        },
        "ref": null,
        "rendered": "4500000000",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "R2-D2",
            "Palpatine",
          ],
        },
        "ref": null,
        "rendered": Array [
          "R2-D2",
          "Palpatine",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "grassy hills, swamps, forests, mountains",
        },
        "ref": null,
        "rendered": "grassy hills, swamps, forests, mountains",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorites",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Naboo
          </h3>,
          <h4 />,
          <h4 />,
          <h4 />,
          <h4>
            temperate
          </h4>,
          <h4>
            4500000000
          </h4>,
          <h4>
            R2-D2
            Palpatine
          </h4>,
          <h4>
            grassy hills, swamps, forests, mountains
          </h4>,
          <h4 />,
          <h4 />,
          <h4 />,
          <button>
            Favorites
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Naboo",
          },
          "ref": null,
          "rendered": "Naboo",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "temperate",
          },
          "ref": null,
          "rendered": "temperate",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "4500000000",
          },
          "ref": null,
          "rendered": "4500000000",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "R2-D2",
              "Palpatine",
            ],
          },
          "ref": null,
          "rendered": Array [
            "R2-D2",
            "Palpatine",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "grassy hills, swamps, forests, mountains",
          },
          "ref": null,
          "rendered": "grassy hills, swamps, forests, mountains",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
